aea/helpers/profiling.py

```diff
@@ -208,18 +219,44 @@ def get_profile_data(self) -> Dict:
         }
 
 
-def get_most_common_objects_in_gc(number: int = 15) -> List[Tuple[str, int]]:
+def get_object_type(obj: Any) -> str:
+    """Returns a string representation of an object's type"""
+    return str(
+        getattr(obj, "__class__", type(obj)).__name__
+    )  # not all objects have the __class__ attribute
+
+
+def get_most_common_objects_in_gc(number: int = 10) -> Dict[str, List[Tuple[str, int]]]:
     """Get the highest-count objects in the garbage collector."""
 
     object_count: CounterType = Counter()
+    object_lengths: CounterType = Counter()
     lock.acquire()
     try:
         for obj in gc.get_objects():
-            object_type = str(
-                getattr(obj, "__class__", type(obj)).__name__
-            )  # not all objects have the __class__ attribute
+            object_type = get_object_type(obj)
             if object_type not in PROFILER_TYPE_BLACK_LIST:
                 object_count[object_type] += 1
+            if type(obj) in (dict, list, set) and len(obj) > 0:
```

Comment:
so any black listed dict, list and set you also want to analyse, correct? The preferred way of checking type is using the built-in `isinstance`: `if obj and isinstance(obj, (list, set, dict)):`